rfcs/text/0000-create-file-macho.md

```diff
@@ -0,0 +1,146 @@
+|--------------------------------------------|------------|-----------------------------------------------------------------------------|
+|   macho.cpu                          |   object   |   CPU information for the file.                                             |
+|   macho.cpu.architecture             |   keyword  |   CPU architecture target for the file.                                     |
+|   macho.cpu.byte_order               |   keyword  |   CPU byte order for the file.                                              |
+|   macho.cpu.subtype                  |   keyword  |   CPU subtype for the file.                                                 |
+|   macho.cpu.type                     |   keyword  |   CPU type for the file.                                                    |
+|   macho.headers                      |   nested   |   Header information for the file.                                          |
+|   macho.headers.commands.number      |   long     |   Number of load commands for the Mach-O header.                            |
+|   macho.headers.commands.size        |   long     |   Size of load commands of the Mach-O header.                               |
+|   macho.headers.commands.type        |   keyword  |   Type of the load commands for the Mach-O header.                          |
+|   macho.headers.magic                |   keyword  |   Magic field of the Mach-O header.                                         |
+|   macho.headers.flags                |   keyword  |   Flags set in the Mach-O header.                                           |
+|   macho.segments                     |   nested   |   Segment information for the file.                                         |
+|   macho.segments.vmaddr              |   keyword  |   Memory address of this segment.                                           |
+|   macho.segments.name                |   keyword  |   Name of this segment.                                                     |
+|   macho.segments.vmsize              |   keyword  |   Memory size of this segment.                                              |
+|   macho.segments.fileoff             |   keyword  |   File offset of this segment.                                              |
+|   macho.segments.filesize            |   keyword  |   Amount of memory to map from the file.                                    |
+|   macho.segments.sections            |   flattened   |   Section information for the segment of the file.                          |
+|   macho.segments.sections.flags      |   keyword  |   Section flags for the segment of the file.                                |
+|   macho.segments.sections.name       |   keyword  |   Section name for the segment of the file.                                 |
+|   macho.segments.sections.type       |   keyword  |   Section type for the segment of the file.                                 |
```

Comment:
Similar to what we discussed https://github.com/elastic/ecs/pull/1077#discussion_r572274291, `flattened` cannot have explicit fields mapped underneath.

If the data is expected to be shaped to an expected object format, we'll have to capture this another way.

Reply:
Do you mean that `macho.segments.sections` needs to be `nested`?

Reply:
No, I mean if we use `flattened` and are expecting specific keys to be present under the `flattened` parent field, we'll need someone to capture this expectation in ECS.

Reply:
It does appear that there are specific keys under `macho.segments.sections.*`. @dcode is this something you can help with?

Here's a reference that could help: https://github.com/aidansteele/osx-abi-macho-file-format-reference#section under `flags`.

Reply:
The fields under `macho.segments.sections[*]` is well-formed and consists of `flags` that denote how the memory is handled by the loader, the name of the section, the section name, and the type of section that it is. In ELF and PE, we put section offset information in `file.*.sections[*]`. This data is missing from the current rendering of the schema. Both the section and the segment have offset data. While in macho, sections always exist within a specific segment, they are actually two different views into the executable and are processed differently at runtime.

Reply:
I suggest we go with using `nested` for `macho.segments.sections.*`. Any objection?

Reply:
I agree with nested, but I think segments and sections should ideally match the usage as in ELF. Namely, we pull out sections to `file.macho.sections` and segments stay at `file.macho.segments`. The data source above nests them and it isn't wrong, but they are two different views into the binary. Sections also have physical offset data.

I'm going to resolve this conversation and suggest those changes (I can't seem to do it with this thread in the middle of it).